### DISCUSSION TITLE
Job/Class limit lacks User feedback and is very restrictive for large servers

### DISCUSSION THREAD

**Discussion by WJohn** [FakeFishGames/Barotrauma#14141] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

1. When trying to join a server whose job limits are already filled out, the game will provide no indication whatsoever that you will be moved away from your desired role. This can be very frustrating on larger servers due to the following:

2. The game instates a class limit for each job which follows this pattern: 1 Captain, 2 Doctor, 2 Security Officers, and unlimited Engineers/Mechanics/Assistants. The problem comes with larger servers, such as 16 players, where being limited to 2 Doctors to patch everyone up or only 2 Security Officers to keep the peace becomes extremely difficult -- and that these limited roles will almost always be already filled by other Users. Newly joining players are frequently prevented from joining the job they wanted.

Ideally, the game should warn the User that the selected job in the Job Preference is currently full (perhaps highlight it in a red outline with a "Job limit reached" displaying). Second to that, an adjustable job limit for server hosts would largely fix this issue on large servers, as right now the only way to do this is through modding.

### Reproduction steps

1. Join a multiplayer server with several other players
2. Ensure that there is already at least 1 Doctor or 2 Security Officers
3. Attempt to join the round as Doctor or Security Officer when those roles are full
4. Observe whether the User is shifted to a different role without notification

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Vykn** [comment]
The limit for doctors is 2, not 1

But yeah annoying thing to have to work around

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/pull/5502 (PR in our private repo). Just ended up removing these restrictions altogether, I agree they're a major inconvenience and don't really serve much purpose.

**Comment by Regalis11** [comment]
The fix has been merged to our private repo now, coming up in a future update!
